Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -31,11 +31,11 @@ public final class AWSClient {
     }
 
     /// Specifies how `EventLoopGroup` will be created and establishes lifecycle ownership.
-    public enum EventLoopGroupProvider {
+    public enum HTTPClientProvider {
         /// `EventLoopGroup` will be provided by the user. Owner of this group is responsible for its lifecycle.
-        case shared(EventLoopGroup)
+        case shared(AWSHTTPClient)
         /// `EventLoopGroup` will be created by the client. When `syncShutdown` is called, created `EventLoopGroup` will be shut down as well.
-        case useAWSClientShared
+        case createNew
```

Comment:
I think, I prefer if the users must bring their own `HTTPClient`. If we create a new one for them, we will need to shut the client down ourselves. Which leads us to a `shutdown` function. But that is only needed if the users uses `createNew`. All I want to say: it becomes messy.

Reply:
There are different HTTP Client requirements for different platforms. Do you expect everyone to know these out of the box? I'm not overly happy with the shutdown process (ie in the deinit) but if we move to the single client I would be happy to move to a syncShutdown() a la AsyncHTTPClient. If the user is required to create their own HTTPClient the shutdown mess is forced onto them.
```
let httpClient = ChooseTheRightClient()
let awsClient = AWSClient(httpClient)
...
httpClient.syncShutdown()
```
vs
```
let awsClient = AWSClient(httpClientProvider: .createNew)
...
awsClient.syncShutdown()
```